File: 백준/Silver/10828. 스택/스택.py
import sys
def push(x):
    lst.append(x)

def pop():
    if len(lst)==0:
        print('-1')
    else:
        print(lst.pop())

def size():
    print(len(lst))

def empty():
    if len(lst)==0:
        print('1')
    else:
        print('0')

def top():
    if len(lst)==0:
        print('-1')
    else:
        print(lst[-1])


lst = []
loop = int(sys.stdin.readline())
# input() 함수는 시간초과 에러 뜸..
# 입출력 속도 비교: sys.stdin.readline() > raw_input() > input()
for i in range(loop):
    lst2 = list(sys.stdin.readline().split())
    if lst2[0] == 'push':
        lst2[1] = int(lst2[1])
        push(lst2[1])
    elif lst2[0] == 'pop':
        pop()
    elif lst2[0] == 'top':
        top()
    elif lst2[0] == 'size':
        size()
    elif lst2[0] == 'empty':
        empty()
    else:
        print('잘못된 입력입니다')

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.ArrayList;
import java.util.StringTokenizer;


public class Main {
    static ArrayList<Integer> list = new ArrayList<Integer>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int loop = Integer.parseInt(br.readLine());
        String msg; int num;

        int i=0;
        while(i<loop) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            msg = st.nextToken();

            switch (msg) {
                case "push":
                    num = Integer.parseInt(st.nextToken());
                    push(num);
                    break;
                case "pop":
                    System.out.println(pop());

                    break;
                case "size":
                    System.out.println(size());
                    break;
                case "empty":
                    System.out.println(empty());
                    break;
                case "top":
                    System.out.println(top());
                    break;
            }
            i++;
        }
        br.close();
    }
    public static void push(int x) {
        list.add(x);
    }
    public static Integer pop() {
        if (list.size() != 0)
            return list.remove(list.size()-1);
        else
            return -1;
    }
    public static Integer size() {
        return list.size();
    }
    public static Integer empty() {
        if (list.size() == 0)
            return 1;
        else
            return 0;
    }
    public static Integer top() {
        if (list.size() != 0)
            return list.get(list.size() - 1);
        else
            return -1;
    }
}
